### PR TITLE
MVKCmdClearImage: Use layered rendering when possible.

### PR DESCRIPTION
**Pull request by cdavis5e**
This makes the clearing tests from the CTS marginally faster, by about 30 ms in many cases.

### CODE REVIEW

**Review by billhollings — APPROVED** [review]
Elegant. Nice idea!

File: MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm [hunk 1]
```diff
@@ -1345,20 +1345,41 @@ static inline MTLSize mvkClampMTLSize(MTLSize size, MTLOrigin origin, MTLSize ma
...
+                    mtlRPSADesc.depthPlane = layerStart;
                 } else {
-                    mtlRPCADesc.slice = layer;
-                    mtlRPDADesc.slice = layer;
-                    mtlRPSADesc.slice = layer;
+                    mtlRPCADesc.slice = layerStart;
+                    mtlRPDADesc.slice = layerStart;
+                    mtlRPSADesc.slice = layerStart;
                 }
+                mtlRPDesc.renderTargetArrayLength = (layerCnt == VK_REMAINING_ARRAY_LAYERS
```
Comment (billhollings) [inline MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm:1360]: @cdavis5e `renderTargetArrayLength` not available on tvOS. Use `renderTargetArrayLengthMVK` instead.

**Review by billhollings — COMMENTED** [review]
Build fail on tvOS.

File: MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm (hunk 1 above)
Comment (cdavis5e) [reply MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm:1360]: #1063

**Pull request merged**